v1/plugin/session.go

```diff
@@ -32,18 +32,29 @@ type Arg struct {
 
 	// enable pprof
 	Pprof bool
+
+	// Path to TLS certificate file for a TLS server
+	CertPath string
+
+	// Path to TLS private key file for a TLS server
+	KeyPath string
+
+	// Flag requesting server to establish TLS channel
+	TLSEnabled bool
 }
 
 // getArgs returns plugin args or default ones
-func getArgs() error {
+func getArgs() (*Arg, error) {
 	pluginArg := &Arg{}
-	if os.Args[1] == "" {
-		return nil
+	osArgs := libInputOutput.readOSArgs()
+	if osArgs[1] == "" {
+		return nil, nil
 	}
-	err := json.Unmarshal([]byte(os.Args[1]), pluginArg)
+	err := json.Unmarshal([]byte(osArgs[1]), pluginArg)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	fmt.Fprintf(os.Stderr, "Got args: %#v\n", pluginArg)
```

Comment:
Having this log here you will have information about plugin's arguments only if there are not nil and unmarshalling was successful. To debug purpose it would be good to have always information about all cases. Please consider it.